### PR TITLE
Functional tests - Add new test 'Search, add and remove voucher from new order page'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'Search, add and remove voucher from new order page'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28159)

### CODE REVIEW

**Review by kpodemski — CHANGES_REQUESTED** [review]
Hello, tests are red

**Review by kpodemski — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/discounts/add.js [hunk 1]
```diff
@@ -257,9 +257,15 @@ class AddCartRule extends BOBasePage {
...
-    return this.getAlertSuccessBlockContent(page);
+    if (waitForNavigation) {
+      // Save and return successful message
+      await this.clickAndWaitForNavigation(page, this.saveButton);
+      return this.getAlertSuccessBlockContent(page);
+    }
+
+    // Save
+    await this.waitForSelectorAndClick(page, this.saveButton);
+    return page;
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/discounts/add.js]:
```suggestion
    return null;
```

No need to return page, i think

File: tests/UI/pages/BO/catalog/discounts/add.js [hunk 2]
```diff
@@ -240,14 +240,14 @@ class AddCartRule extends BOBasePage {
...
   }
 
-
   /**
    * Create/edit cart rule
    * @param page {Page} Browser tab
    * @param cartRuleData {CartRuleData} Data to set on add/edit cart rule form
-   * @returns {Promise<string>}
+   * @param waitForNavigation {boolean} True if we need to save and waitForNavigation
+   * @returns {Promise<string|*>}
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/discounts/add.js]:
```suggestion
   * @returns {Promise<string|null>}
```

**Review by boubkerbribri — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Pull request merged**